server/Arcadia.Assistant.CSP/CspCalendarEventsApprovalsChecker.cs

```diff
@@ -7,26 +7,62 @@
 
     using Akka.Actor;
 
-    using Organization.Abstractions;
+    using Arcadia.Assistant.Calendar.Abstractions;
+    using Arcadia.Assistant.Organization.Abstractions;
 
-    public class CspVacationApprovalsChecker : VacationApprovalsChecker
+    public class CspCalendarEventsApprovalsChecker : CalendarEventsApprovalsChecker
     {
         private const string DepartmentsStorageActorPath = @"/user/organization/departments/departments-storage";
         private const string EmployeesStorageActorPath = @"/user/organization/employees/employees-storage";
 
         private readonly Regex sdoDepartmentRegex = new Regex(@"SDO\..+\..+");
 
-        protected override async Task<string> GetNextApprover(string employeeId, IEnumerable<string> existingApprovals)
+        protected override async Task<string> GetNextApprover(string employeeId, IEnumerable<string> existingApprovals, string eventType)
         {
             var employeesActor = Context.ActorSelection(EmployeesStorageActorPath);
             var departmentsActor = Context.ActorSelection(DepartmentsStorageActorPath);
 
             var allDepartments = await this.GetDepartments(departmentsActor);
-            var employeeDepartmentId = await this.GetEmployeeDepartmentId(employeesActor, employeeId);
+            var employeeMetadata = await this.GetEmployeeDepartmentId(employeesActor, employeeId);
```

Comment:
employeeMetadata = GetEmployeeDepartmentId? :)

Reply:
aslo you never check employeeMetadata for null

Reply:
Shit, I missed this. It was too late...

Reply:
Fixed both.